pkg/tokenauth/sources/oidc/oidc.go

```diff
@@ -43,11 +43,33 @@ func NewWithCache(name string, cache login.Cache, callbackSrv *login.CallbackSer
 
 	return &source{
 		name:        name,
-		cache:       cache,
 		tokenSource: tokenSource,
 	}, clearIDTokenFunc, nil
 }
 
+// NewGoogleFromServiceAccount constructs tokenauth.Source that is able to return valid OIDC token from given Google Service Account.
+func NewGoogleFromServiceAccount(name string, config login.OIDCConfig, googleServiceAccountJSON []byte) (tokenauth.Source, error) {
+	tokenSource, _, err := gsa.NewOIDCTokenSource(
+		context.Background(),
+		log.New(os.Stdout, "", log.LstdFlags),
+		googleServiceAccountJSON,
+		config.Provider,
+		gsa.OIDCConfig{
+			ClientID:     config.ClientID,
+			ClientSecret: config.ClientSecret,
+			Scopes:       config.Scopes,
+		},
+	)
+	if err != nil {
+		return nil, err
```

Comment:
Can you wrap this error and add more info here?

Reply:
yup